scripts/index.js

```diff
@@ -1 +1,158 @@
+
+//is positive with filter
+function positiveFilter(num){
+    if (num >= 0){
+        return num;
+    }
+}
+console.log("2nd variation of positive numbers using filter method.  this will not return zero")
+const numbers = [1, 2, 4, -2, -1, 0, 5, -9, -6, -9.8, 8.7, 3];
+console.log(numbers.filter(positiveFilter));
+
+function evenNumbers(ar) {
+    const newArray = []
+    ar.forEach(function(num){
+        if ((num % 2) === 0) {
+            newArray.push(num);
+        }
+    })
+    return newArray
+}
+console.log (evenNumbers([1, -4, 5, 2.2, -3, 0, 11, 10, 27, 3.3]));
+
+function squareTheNumbers(ar) {
+   return ar.map(function(num){
+        return num * num;
+    });
+}
+
+console.log(squareTheNumbers([1,2,3,4]));
+
+const cities = [
+    { name: 'Los Angeles', temperature: 60.0},
+    { name: 'Atlanta', temperature: 52.0 },
+    { name: 'Detroit', temperature: 48.0 },
+    { name: 'New York', temperature: 80.0 }
+  ];
+
+//   console.log(cities);
+
+function citiesBelow70(allcities) {
+    const newArray = []; 
+
+    allcities.forEach(function(city){
+        if (city.temperature < 70) {
+            
+            newArray.push(city.name);
+        }
+    });
+    return newArray;
+}
+console.log(citiesBelow70(cities));
+
+function cityNames(allcities) {
+    return allcities.map(function(city) {
+        return city.name;
+    });
+}
+console.log(cityNames(cities));
+
+var people = [
+    'Dom',
+    'Lyn',
+    'Kirk',
+    'Autumn',
+    'Trista',
+    'Jesslyn',
+    'Kevin',
+    'John',
+    'Eli',
+    'Juan',
+    'Robert',
+    'Keyur',
+    'Jason',
+    'Che',
+    'Ben'
+  ];
+
+  function goodJob(names) {
+      names.forEach(function(each){
+            console.log(`Good Job, ${each}!`);
+      })
+  }
+
+  goodJob(people);
+
+  function helloWorld(){
+    console.log("Hello, world!");
+}
+
+  function call3Times(fun) {
+    fun();
+    fun();
+    fun();
+  }
+
+call3Times(helloWorld);
+
+function callNTimes(afunc,num){
+    for (i = 1; i<=num; i++){
+        afunc();
+    }
+}
+
+callNTimes(helloWorld,6);
+
+//given
+//print a string x times without using a for loop
+function range(min, max) {
+    var arr = [];
+    for (var i = min; i < max; i++) {
+      arr.push(i);
+    }
+    return arr;
+  }
+
+function strMultiply(str,times) {
+    counterArray = range(1,times);      //[1,2,3,4,5]
+    newstring = "";
+    counterArray.forEach(function(){
+        newstring += str;
+    });
+    return newstring;
+}
+console.log(strMultiply('abc', 5));
+
+//sort alphabetically by name
+console.log(people.sort());
+
+//sort by length of name
+people.sort(function(a,b){
+   return  a.length - b.length;
+});
+
+console.log(people);
+
+//sort by price
+var products = [
+    { name: 'Basketball', price: 12.00 },
+    { name: 'Tennis Racquet', price: 66.00 },
+    { name: 'Tennis Balls', price: 9.00 },
+    { name: 'Tennis Balls', price: 9.00 }
+  ];
+
+console.log(products.sort(function(a,b){
```

Comment:
Can you think of a way to implement a function that you could call like this?

`by('price')` or `by('length')`

So that you could do `products.sort(by('price'))` or `people.sort(by('length'))`?

It's much more of an advanced technique, but very useful.
See if some googling ("javascript return function from function") might lead you towards some ideas.